katas/metal-slug/samuel/Soldier.js

```diff
@@ -0,0 +1,24 @@
+import { Civilian } from './Civilian.js';
+import { Enemy } from './Enemy.js';
+
+export class Soldier {
+    constructor() {
+        this.resistance = 1;
+        this.attackPower = 0;
+        this.lifes = 3;
+    }
+    changeWeapon(weaponAttackPower) {
+        this.attackPower = 1;
+        this.attackPower = this.attackPower + weaponAttackPower;
+        return this.attackPower;
+    }
+    useKnife(civilianOrEnemy) {
+        if (civilianOrEnemy instanceof Civilian) {
```

Comment:
This code is ok, but maybe can tend to add too much responsability in case we have more than twos types of target? say, we want to add (besides civilian and enemy) box to one of the elements that can interact with "knife" action of the soldier. We can add a new "if" and set the new behavior, but this is adding more responsability to this function and going to open-close principle at the same time